homeassistant/core.py

```diff
@@ -1049,9 +1049,17 @@ def async_entity_ids_count(
 
     def all(self, domain_filter: str | Iterable | None = None) -> list[State]:
         """Create a list of all states."""
-        return run_callback_threadsafe(
```

Comment:
Why change this? You call it once every hour.

Reply:
Yeah, it's a little bit meh maybe.
This change was suggested here: https://github.com/home-assistant/core/pull/49852#discussion_r626005752

Reply:
Reverted in b0ae32a2bc14ec4c7e9be437a0b6f21c17a2bac8, we can discuss potential optimizations of `StateMachine` methods separately.